### PR TITLE
Refresh the development enviroment, care about missed "verify_ssl_cert=True" and stabilize/fix doctests in "https.txt"

### PR DESCRIPTION
**Pull request by amotl**
Hi there,

when using Python 3.9, setting up the development environment croaked. Because `bootstrap.py` was a workaround in pre-`pip` times [1], and to modernize this repository, this patch completely gets rid of it. On the other hand, it pins to appropriate package versions to still be compatible with Python 3.5 through Python 3.9, where the former setup didn't satisfy these requirements anymore.

Along the lines, the patch fixes the doctests in `https.txt` in order to improve the situation with #375. For doing this appropriately, it takes care about a missed `verify_ssl_cert=True` setting on the HTTP client as a followup to #385.

We need this to be able to run #400 successfully on CI.

With kind regards,
Andreas.

[1] https://community.plone.org/t/not-using-bootstrap-py-as-default/620

### CODE REVIEW

**Comment by amotl** [comment]
> slightly improves the test layer setup machinery as an attempt to improve the situation with #375

We are not there yet. Because this error is still there, this patch switches to `fail-fast: false` on the test matrix in order to at least run it on all available Python versions instead of failing the complete suite immediately after tripping into the `https.txt` problem.

However, because of [1], we need this patch to be integrated before being able to run #400 on CI.

[1] https://github.com/crate/crate-python/pull/400/checks?check_run_id=2133451691#step:4:575

**Comment by mfussenegger** [comment]
It's a bit hard to review as this PR does several things. Any chance to break this up into multiple smaller ones?

**Comment by amotl** [comment]
> It's a bit hard to review as this PR does several things. Any chance to break this up into multiple smaller ones?

I can do it. But I will probably not be able to make CI succeed on individual chunks because currently, it is broken due to both the environment being outdated because of issues with `buildout` and the nasty `https.txt` woes on the other side of the story. However, I can try to break out 06067a2, 4eeff6f, a63e702 and d2fb2b5 into a different patch, or, on the other hand, hand in 1081140 separately.

**Comment by amotl** [comment]
Hi @mfussenegger and @seut,

I removed some of the commits here in order to boil it down to the minimum required things to make CI succeed again. I will submit a separate patch including them.

With kind regards,
Andreas.

File: DEVELOP.rst [hunk 1]
```diff
@@ -1,46 +1,57 @@
...
 This project uses buildout_ to set up the development environment.
 
-To start things off, run::
+To start things off, create a Python virtualenv and install buildout::
 
-    $ python bootstrap.py
+    python3 -m venv .venv
+    source .venv/bin/activate
+
+    # Workaround for Python 3.9
```
Comment (seut) [inline DEVELOP.rst:15]:
```suggestion
    # Workaround for Python 3.5
```

**Review by seut — APPROVED** [review]
Looks good otherwise.

File: DEVELOP.rst (hunk 1 above)
Comment (amotl) [reply DEVELOP.rst]: Thanks for spotting this.

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**